Nodejs/Product/Nodejs/Project/NodejsFileNode.cs

```diff
@@ -35,14 +35,16 @@ public NodejsFileNode(NodejsProjectNode root, ProjectElement e)
         }
 
         protected override void OnParentSet(HierarchyNode parent) {
-            if (ProjectMgr == null || ProjectMgr.Analyzer == null || !ProjectMgr.ShouldAcquireTypingsAutomatically) {
+            if (ProjectMgr == null || ProjectMgr.Analyzer == null) {
                 return;
             }
 
             if (Url.EndsWith(NodejsConstants.TypeScriptDeclarationExtension, StringComparison.OrdinalIgnoreCase) && Url.IndexOf(@"\typings\", StringComparison.OrdinalIgnoreCase) >= 0) {
-                ProjectMgr.Site.GetUIThread().Invoke(() => {
-                    this.IncludeInProject(true);
-                });
+                if (ProjectMgr.ShouldAcquireTypingsAutomatically) {
```

Comment:
This hang happens when the value is first needed, so caching will not fit it.

This reordering is not needed for the fix, but ensures that even if the other part of the change does not fix the hang, we would only run into the hang when creating a project from code with an existing `typings` directory.

The actual operation `IsTypeScriptProject` is not all that expensive when called at the correct time.